From 0f46a6bcafdecdc3c3808853356b445279e08880 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Fri, 14 Dec 2018 17:38:12 +0800
Subject: [PATCH 2/3] Fix carrier config option not hidden on a CDMA phone

 * The option is added twice, from GsmUmtsOptions and CdmaOptions, and
   GsmUmtsOptions unconditionally adds it if current phone type is not
   GSM. This is obviously wrong, and on a CDMA phone the option persists
   even if it's disabled from CarrierConfig. Once a user selects it,
   phone app will crash.
 * To fix it, reverse the initial value of the boolean, so that
   GsmUmtsOptions won't add the option if current phone type is not GSM.

Change-Id: I5a9939bf43a4c17807a0171861a1a76029f5379d
---
 src/com/android/phone/GsmUmtsOptions.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/phone/GsmUmtsOptions.java b/src/com/android/phone/GsmUmtsOptions.java
index 19cd3ef4c..b49de87e2 100644
--- a/src/com/android/phone/GsmUmtsOptions.java
+++ b/src/com/android/phone/GsmUmtsOptions.java
@@ -71,7 +71,7 @@ public class GsmUmtsOptions {
     protected void update(final int subId, INetworkQueryService queryService) {
         boolean addAPNExpand = true;
         boolean addNetworkOperatorsCategory = true;
-        boolean addCarrierSettings = true;
+        boolean addCarrierSettings = false;
         Phone phone = PhoneGlobals.getPhone(subId);
         if (phone == null) return;
         if (phone.getPhoneType() != PhoneConstants.PHONE_TYPE_GSM) {
-- 
2.17.1

